src/main.lua

```diff
@@ -2695,71 +2725,70 @@ addEventingHandler("auctionBid", utils.hasMatchingTag("Action", ActionMap.Auctio
 				years = years or 1
 			end
 		end
-	end
 
-	local validateStatus, errorMessage = pcall(checkAssertions)
-	if not validateStatus then
-		ao.send({
-			Target = msg.From,
-			Action = "Invalid-" .. ActionMap.AuctionBid .. "-Notice",
-			Error = "Bad-Input",
-			Data = tostring(errorMessage),
-		})
-		return
+		local auction = arns.getAuction(name)
+		assert(auction, "Auction not found")
 	end
 
-	local auction = arns.getAuction(name)
-	if not auction then
+	local shouldContinue = eventingPcall(msg.ioEvent, function(error)
 		ao.send({
 			Target = msg.From,
-			Action = "Invalid-" .. ActionMap.AuctionBid .. "-Notice",
-			Error = "Auction-Not-Found",
+			Tags = { Action = "Invalid-" .. ActionMap.AuctionBid .. "-Notice", Error = "Bad-Input" },
+			Data = tostring(error),
 		})
+	end, checkAssertions)
+	if not shouldContinue then
 		return
 	end
 
-	local status, auctionBidOrError =
-		pcall(arns.submitAuctionBid, name, bidAmount, bidder, timestamp, processId, type, years)
-	if not status then
+	local shouldContinue2, result = eventingPcall(msg.ioEvent, function(error)
 		ao.send({
 			Target = msg.From,
-			Action = "Invalid-" .. ActionMap.AuctionBid .. "-Notice",
```

Comment:
Is this the incorrect way to do this or is there just an advantage to setting these to Tags manually?

Reply:
we have some incosistency accross notices that this is addressing